proxies/aareg-proxy/src/main/java/no/nav/testnav/proxies/aaregproxy/AaregProxyApplicationStarter.java

```diff
@@ -18,69 +18,82 @@
-            @Value("${sts.preprod.token.provider.username}") String username,
-            @Value("${sts.preprod.token.provider.password}") String password
+        @Value("${sts.preprod.token.provider.url}") String url,
+        @Value("${sts.preprod.token.provider.username}") String username,
+        @Value("${sts.preprod.token.provider.password}") String password
     ) {
         return new StsOidcTokenService(url, username, password);
     }
 
     @Bean
     public StsOidcTokenService stsTestOidcTokenService(
-            @Value("${sts.test.token.provider.url}") String url,
-            @Value("${sts.test.token.provider.username}") String username,
-            @Value("${sts.test.token.provider.password}") String password
+        @Value("${sts.test.token.provider.url}") String url,
+        @Value("${sts.test.token.provider.username}") String username,
+        @Value("${sts.test.token.provider.password}") String password
     ) {
         return new StsOidcTokenService(url, username, password);
     }
 
     @Bean
     public RouteLocator customRouteLocator(RouteLocatorBuilder builder, StsOidcTokenService stsTestOidcTokenService, StsOidcTokenService stsPreprodOidcTokenService) {
+        RouteLocatorBuilder.Builder routes = builder.routes();
 
         var preprodFilter = AddAuthenticationRequestGatewayFilterFactory
-                .bearerAuthenticationAndNavConsumerTokenHeaderFilter(stsPreprodOidcTokenService::getToken);
+            .bearerAuthenticationAndNavConsumerTokenHeaderFilter(stsPreprodOidcTokenService::getToken);
+        Stream.of("q0", "q1", "q2", "q4", "q5", "qx")
+            .forEach(env -> routes
+                .route(createRoute(env, preprodFilter))
+                .route(env, createQueryBasedRoute(env, preprodFilter)));
+
         var testFilter = AddAuthenticationRequestGatewayFilterFactory
-                .bearerAuthenticationAndNavConsumerTokenHeaderFilter(stsTestOidcTokenService::getToken);
-        return builder
-                .routes()
-                .route(createRoute("q0", preprodFilter))
-                .route(createRoute("q1", preprodFilter))
-                .route(createRoute("q2", preprodFilter))
-                .route(createRoute("q4", preprodFilter))
-                .route(createRoute("q5", preprodFilter))
-                .route(createRoute("qx", preprodFilter))
-                .route(createRoute("t0", testFilter))
-                .route(createRoute("t1", testFilter))
-                .route(createRoute("t2", testFilter))
-                .route(createRoute("t3", testFilter))
-                .route(createRoute("t4", testFilter))
-                .route(createRoute("t5", testFilter))
-                .route(createRoute("t6", testFilter))
-                .route(createRoute("t13", testFilter))
-                .build();
+            .bearerAuthenticationAndNavConsumerTokenHeaderFilter(stsTestOidcTokenService::getToken);
+        Stream.of("t0", "t1", "t2", "t3", "t4", "t5", "t6", "t13")
```

Comment:
jeg tror T0 er også prodlikt miljø (eller Pensjon bruker T0 for reproduserer feil fra Prod)